Snackbar Font

VladislavAntonyuk opened the pull request:
### Description of Change ###

Use Font instead of FontFamily and FontSize

![image](https://user-images.githubusercontent.com/33021114/100662125-245d4700-335d-11eb-9bd7-346a780e316b.png)

![image](https://user-images.githubusercontent.com/33021114/100779779-8b3c3800-3411-11eb-853b-5393b4a0fb25.png)



### Bugs Fixed ###

- Fixes #607

### API Changes ###

Added: 
 
- `Font Font { get; set; }`
- static Font DefaultFont`

Removed:

 - `FontFamily`
 - `FontSize`

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [x] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

pictos commented:
On UWP I'm facing this exception, when I press any button

![image](https://user-images.githubusercontent.com/20712372/100673825-e81ced00-3342-11eb-8c15-84664f9ad6fa.png)

This one for WPF
![image](https://user-images.githubusercontent.com/20712372/100674071-582b7300-3343-11eb-9c53-08d3bdb19bd0.png)

On Android
When I tap on the first two button, nothing happens. The last one works as expected.

On iOS
There is no background for the first and second buttons.
![image](https://user-images.githubusercontent.com/20712372/100675143-26b3a700-3345-11eb-821a-6d05252568d6.png)

VladislavAntonyuk commented:
> @VladislavAntonyuk thanks for this PR, I found some issues when I ran it on my end.
> 
> 
> 
> On UWP I'm facing this exception, when I press any button
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/20712372/100673825-e81ced00-3342-11eb-8c15-84664f9ad6fa.png)
> 
> 
> 
> This one for WPF
> 
> ![image](https://user-images.githubusercontent.com/20712372/100674071-582b7300-3343-11eb-9c53-08d3bdb19bd0.png)
> 
> 
> 
> On Android
> 
> When I tap on the first two button, nothing happens. The last one works as expected.
> 
> 
> 
> On iOS
> 
> There is no background for the first and second buttons.
> 
> ![image](https://user-images.githubusercontent.com/20712372/100675143-26b3a700-3345-11eb-821a-6d05252568d6.png)
> 
> 
> 
> 

Oops, found the issue. In the latest commit I changed Font property value to Font.Default. I will rollback it to Font.OfSystemFontSize(14)

AndreiMisiukevich commented on src/CommunityToolkit/Xamarin.CommunityToolkit.Sample/Xamarin.CommunityToolkit.Sample.csproj:
```diff
@@ -31,16 +41,5 @@
-    </Compile>
-    </ItemGroup>
 </Project>
 
```
Can we discard this file?

VladislavAntonyuk replied:
looks like it can be removed.

AndreiMisiukevich commented on src/CommunityToolkit/Xamarin.CommunityToolkit.Sample/Pages/Views/SnackBarPage.xaml.cs:
```diff
@@ -35,9 +35,8 @@ async void DisplaySnackBarAdvancedClicked(object sender, EventArgs args)
-				FontFamily = "Times New Roman",
-				Message = GenerateLongText(5)
+				Font = Font.OfSize("FARegular", 40),
+				Message = "\uf2bb"
```
I propose to convert font to class with something like https://andreinitescu.github.io/IconFont2Code/
And use constants instead :)

VladislavAntonyuk replied:
I added this code just for example. Do you think it is worth to add all icons? Maybe create a constant just for this text?

AndreiMisiukevich replied:
Would be enough it, agreed 👍

VladislavAntonyuk commented on src/CommunityToolkit/Xamarin.CommunityToolkit.Sample/Xamarin.CommunityToolkit.Sample.csproj:
```diff
@@ -10,15 +10,7 @@
     <EmbeddedResource Remove="resources\**" />
     <None Remove="resources\**" />
     <EmbeddedResource Include="Images\**" />
```
Maybe `<EmbeddedResource Include="Fonts\**" />`?

AndreiMisiukevich replied:
Yes, that's what I meant)

pictos commented:
I can review it by tomorrow. @VladislavAntonyuk this commits also fixed the iOS implementation?

VladislavAntonyuk commented:
> I can review it by tomorrow. @VladislavAntonyuk this commits also fixed the iOS implementation?

yes, it was a Color.Default for the background. I changed it to White

jfversluis commented:
> > I can review it by tomorrow. @VladislavAntonyuk this commits also fixed the iOS implementation?
> 
> yes, it was a Color.Default for the background. I changed it to White

How does this work with Dark Mode? Should it use the theme aware colors?

VladislavAntonyuk commented:
> > > I can review it by tomorrow. @VladislavAntonyuk this commits also fixed the iOS implementation?
> 
> > 
> 
> > yes, it was a Color.Default for the background. I changed it to White
> 
> 
> 
> How does this work with Dark Mode? Should it use the theme aware colors?

I think it should, but it is not implemented.
I think we may not set the color at all allowing the system to use theme default color.
Should I implement it in this PR or we need a new issue?

VladislavAntonyuk commented:
with the changes of colors, I also added multiple action buttons for ios and macos. so now they work the same as other platforms

pictos reviewed (approved):
Works as expected on my end.

Tommigun1980 commented:
Hi @VladislavAntonyuk.

After updating the following stopped working as the properties were removed:
```xaml
MessageOptions.DefaultFontSize = (OnPlatform<double>)baseFontSize;
MessageOptions.DefaultFontFamily = (string)snackBarFontFamily;
SnackBarActionOptions.DefaultFontSize = (OnPlatform<double>)baseFontSize;
SnackBarActionOptions.DefaultFontFamily = (string)baseFontFamily;
```

BaseFontSize and BaseFontFamily are used in every component where a font is relevant. Labels for example have styles that set their FontSize and FontFamily from them. This used to be the case for SnackBars as well, but these properties no longer exist.

So this change makes SnackBar completely different from **any** other component that displays text. I liked that I could style SnackBars exactly as I could any other component (albeit from code instead of implicit styling) so I am not too fond of this change. Tbh I don't even know how to create a Font object to assign there as it's not been needed in the public interface of any other component. Is there a reason for complicating this and making this non-standard? I can make a report about this but thought this might belong in this pr.

Thanks.

VladislavAntonyuk commented:
Font struct (https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.font) gives you much more flexibility to customize the components.
Also XF has extension methods which allows to convert the XF font to the Native font. Because we depends on XF I do not see any reason not to use this awesome struct 😄 and just duplicate its properties.

Tommigun1980 commented:
> Font struct (https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.font) gives you much more flexibility to customize the components.
> Also XF has extension methods which allow s to convert the XF font to the Native font. Because we depends on XF I do not see any reason not to use this awesome struct 😄 and just duplicate its properties.

It’s just unstandard as no other components use it in the public api, that’s all.

VladislavAntonyuk commented:
Snackbar is not a component which you usually add in xaml
